File: dotNet/src/User/UserSession.cs
using src.Exception;

namespace src.User;

public class UserSession
{
    private static readonly UserSession userSession = new UserSession();

    private UserSession()
    {
    }

    public static UserSession GetInstance()
    {
        return userSession;
    }

    public src.User.User? GetLoggedUser()
    {
        throw new NotTestableCallException("UserSession.GetLoggedUser() should not be called in an unit test");
    }
}

File: typescript/src/User/UserSession.ts
import {User} from './User';
import {NotTestableCallError} from '../Error/NotTestableCallError';

export class UserSession {
    private static readonly userSession: UserSession = new UserSession();

    private constructor() {
    }

    public static getInstance(): UserSession {
        return UserSession.userSession;
    }

    public getLoggedUser(): User | undefined {
        throw new NotTestableCallError('UserSession.getLoggedUser() should not be called in an unit test');
    }
}
